framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -259,30 +291,31 @@ public AnnotationMirror leastUpperBound(AnnotationMirror a1, AnnotationMirror a2
             }
             // If both are the same type, determine the type and merge:
             else if (AnnotationUtils.areSameIgnoringValues(a1, a2)) {
-                List<Object> a1Values =
-                        AnnotationUtils.getElementValueArray(a1, "value", Object.class, true);
-                List<Object> a2Values =
-                        AnnotationUtils.getElementValueArray(a2, "value", Object.class, true);
-                HashSet<Object> newValues = new HashSet<Object>(a1Values.size() + a2Values.size());
+                if (AnnotationUtils.areSameByClass(a1, IntRange.class)) {
```

Comment:
Above, the `if (!AnnotationUtils.areSameIgnoringValues(...))` test seems wrong.  If both are `@IntRange`, then their lub should not be top.  I suspect I'm missing something here; can you clarify why the code is correct, or fix it if it is not?